### PR TITLE
#3256 Provider Scaling out Interface-Add all apis

### PR DESCRIPTION
**Pull request by ssxlulu**
For #3256.

Changes proposed in this pull request:
- Add all related apis

### CODE REVIEW

File: sharding-distribution/sharding-scaling-distribution/src/main/resources/bin/start.sh [hunk 1]
```diff
@@ -20,8 +20,10 @@ cd `dirname $0`
...
 
+JAVA_OPTS=" -Dport=8080 "
+
 CLASS_PATH=.:${DEPLOY_DIR}/conf:${DEPLOY_DIR}/lib/*
 
 MAIN_CLASS=org.apache.shardingsphere.shardingscaling.Bootstrap
 
-java -classpath ${CLASS_PATH} ${MAIN_CLASS}
\ No newline at end of file
+java ${JAVA_OPTS} -classpath ${CLASS_PATH} ${MAIN_CLASS}
```
Comment (KomachiSion) [inline sharding-distribution/sharding-scaling-distribution/src/main/resources/bin/start.sh]: Moving port to configFile maybe better. You can refer `Bootstrap in sharding-proxy`.

File: sharding-scaling/sharding-scaling-bootstrap/src/main/java/org/apache/shardingsphere/shardingscaling/Bootstrap.java [hunk 2]
```diff
@@ -49,72 +41,36 @@
 @Slf4j
 public class Bootstrap {
 
+    private static final int PORT = Integer.parseInt(System.getProperty("port", "8080"));
+
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-bootstrap/src/main/java/org/apache/shardingsphere/shardingscaling/Bootstrap.java]: Moving port to configFile maybe better. You can refer `Bootstrap in sharding-proxy`.

File: sharding-scaling/sharding-scaling-core/src/test/resources/config.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        "url": "jdbc:mysql://127.0.0.1:3306/test2?serverTimezone=UTC&useSSL=false",
+        "username": "root"
+      }
+  },
+  "serverConfiguration": {
+    "blockQueueSize": 10000,
+    "pushTimeout": 1000,
+    "concurrency": 3
+  }
+}
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-core/src/test/resources/config.json]: Please end file with blank line.

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/web/entity/Job.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * Job entity for http server.
+ *
+ * @author ssxlulu
+ */
+@AllArgsConstructor
+@Data
+public class Job {
+    private int id;
+}
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/web/entity/Job.java]: Can you replace this class by `ShardingScalingJob`?

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/execute/executor/channel/MemoryChannel.java [hunk 5]
```diff
@@ -35,9 +34,9 @@
  */
 public class MemoryChannel implements Channel {
 
-    private static final int PUSH_TIMEOUT = ScalingContext.getInstance().getServerConfiguration().getPushTimeout();
+    private static final int PUSH_TIMEOUT = 1000;
 
-    private final BlockingQueue<Record> queue = new ArrayBlockingQueue<>(ScalingContext.getInstance().getServerConfiguration().getBlockQueueSize());
+    private final BlockingQueue<Record> queue = new ArrayBlockingQueue<>(10000);
```
Comment (KomachiSion) [inline sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/execute/executor/channel/MemoryChannel.java]: Why do you use hardcode in there?

File: sharding-scaling/sharding-scaling-bootstrap/src/main/java/org/apache/shardingsphere/shardingscaling/Bootstrap.java (hunk 2 above)
Comment (ssxlulu) [reply sharding-scaling/sharding-scaling-bootstrap/src/main/java/org/apache/shardingsphere/shardingscaling/Bootstrap.java]: I will submit a new issue for config related, so here will stay as it is?

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/web/entity/Job.java (hunk 4 above)
Comment (ssxlulu) [reply sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/web/entity/Job.java]: Yes, it is a good idea.

File: sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/execute/executor/channel/MemoryChannel.java (hunk 5 above)
Comment (ssxlulu) [reply sharding-scaling/sharding-scaling-core/src/main/java/org/apache/shardingsphere/shardingscaling/core/execute/executor/channel/MemoryChannel.java]: I will submit a new issue for config related.

**Pull request merged**